cmd/juju/model/destroy.go

```diff
@@ -156,19 +181,78 @@ func (c *destroyCommand) Run(ctx *cmd.Context) error {
 	}
 	defer api.Close()
 
+	// Attempt to connect to the controller API.  If we can't, fail the destroy.
+	controllerAPI, err := c.getControllerAPI()
```

Comment:
Not strong enough to block this PR, but other CLI commands being crappy isn't a good argument to make another one crappy. It can be fixed later.